File: test_samples/vulnerable_code.java
"""
Java crypto patterns for testing
"""
import java.security.KeyPairGenerator;
import java.security.KeyPair;
import java.security.MessageDigest;
import java.security.Signature;
import javax.crypto.KeyGenerator;

public class VulnerableCode {
    
    // RSA usage (quantum-vulnerable)
    public KeyPair generateRSAKey() throws Exception {
        KeyPairGenerator keyGen = KeyPairGenerator.getInstance("RSA");
        keyGen.initialize(2048);
        return keyGen.generateKeyPair();
    }
    
    // ECC usage (quantum-vulnerable)
    public KeyPair generateECCKey() throws Exception {
        KeyPairGenerator keyGen = KeyPairGenerator.getInstance("EC");
        keyGen.initialize(256);
        return keyGen.generateKeyPair();
    }
    
    // ECDSA usage (quantum-vulnerable)
    public byte[] ecdsaSign(byte[] data, PrivateKey privateKey) throws Exception {
        Signature signature = Signature.getInstance("SHA256withECDSA");
        signature.initSign(privateKey);
        signature.update(data);
        return signature.sign();
    }
    
    // Weak hash
    public String weakHashSHA1(String input) throws Exception {
        MessageDigest md = MessageDigest.getInstance("SHA-1");
        byte[] hash = md.digest(input.getBytes());
        return bytesToHex(hash);
    }
    
    public String weakHashMD5(String input) throws Exception {
        MessageDigest md = MessageDigest.getInstance("MD5");
        byte[] hash = md.digest(input.getBytes());
        return bytesToHex(hash);
    }
    
    // AES (quantum-resistant for now)
    public SecretKey generateAESKey() throws Exception {
        KeyGenerator keyGen = KeyGenerator.getInstance("AES");
        keyGen.init(256);
        return keyGen.generateKey();
    }
    
    private String bytesToHex(byte[] bytes) {
        StringBuilder result = new StringBuilder();
        for (byte b : bytes) {
            result.append(String.format("%02x", b));
        }
        return result.toString();
    }
}

File: test_samples/vulnerable_code.py
"""
Sample vulnerable code for testing the scanner
"""

from Crypto.PublicKey import RSA
from Crypto.Cipher import PKCS1_OAEP
from cryptography.hazmat.primitives.asymmetric import ec
from cryptography.hazmat.primitives import hashes
import hashlib

# RSA usage (quantum-vulnerable)
def generate_rsa_key():
    key = RSA.generate(2048)
    return key

def rsa_encrypt(data, public_key):
    cipher = PKCS1_OAEP.new(public_key)
    return cipher.encrypt(data)

# ECC usage (quantum-vulnerable) 
def generate_ecc_key():
    private_key = ec.generate_private_key(ec.SECP256R1())
    return private_key

def ecdsa_sign(message, private_key):
    signature = private_key.sign(message, ec.ECDSA(hashes.SHA256()))
    return signature

# Weak hash (partially quantum-vulnerable)
def weak_hash_sha1(data):
    return hashlib.sha1(data.encode()).hexdigest()

def weak_hash_md5(data):
    return hashlib.md5(data.encode()).hexdigest()

# Strong symmetric crypto (quantum-resistant for now)
def aes_encrypt(data, key):
    from Crypto.Cipher import AES
    cipher = AES.new(key, AES.MODE_GCM)
    ciphertext, tag = cipher.encrypt_and_digest(data)
    return ciphertext, tag

# Safe function without crypto
def safe_function():
    return "This function doesn't use cryptography"
